06-reactjs/components/movie-form.js

```diff
@@ -0,0 +1,88 @@
+var React = require('react');
+var Input = require('./input');
+var Button = require('./button');
+var _ = require('lodash');
+
+
+var MovieForm = React.createClass({
+
+    propTypes: {
+        onMovieSubmit: React.PropTypes.func
+    },
+
+    getInitialState: function () {
+        return {
+            title: '',
+            genre: '',
+            rating: '',
+        };
+    },
+
+    render: function () {
+        return (
+            <form {...this.getFormProps()}>
+                {this.renderInputs()}
+                <input type="submit" value="Post" />
+            </form>
+        );
+    },
+
+    renderInputs: function () {
```

Comment:
I'm not sure if I'm convinced with this approach... Why not put each input you need directly within your ``render`` method?
Using a state to iterate and push your components doesn't seems to be something that's going to make things easier in the future. For example, I think the issue you have regarding binding ``this`` is not something you're going to be worried if you do your define your components without mapping them. Also, tying your state to rendering purposes (in this case) doesn't seems to be different than using a common array (and not having your state polluted needlessly).
What you've done is going to work, but seems a bit contrived to me.